common/djangoapps/third_party_auth/tests/test_provider.py

```diff
@@ -75,6 +78,22 @@ def test_enabled_returns_list_of_enabled_providers_sorted_by_name(self):
         with patch('third_party_auth.provider._PSA_OAUTH2_BACKENDS', backend_names):
             self.assertEqual(sorted(provider_names), [prov.name for prov in provider.Registry.enabled()])
 
+    def test_enabled_doesnt_query_site(self):
+        """Regression test for 1+N queries for django_site (ARCHBOM-1139)"""
+        re_from_ds = re.compile(r'FROM\s+"django_site"')
+
+        self.enable_saml()
+        for i in range(100):
+            self.configure_saml_provider(enabled=True, slug="saml-slug-%s" % i)
+
+        with CaptureQueriesContext(connections[DEFAULT_DB_ALIAS]) as cq:
+            enabled_slugs = {p.slug for p in provider.Registry.enabled()}
+
+        self.assertEqual(len(enabled_slugs), 100)
+        # Should not involve any queries for Site, or at least should not *scale* with number of providers
+        django_site_queries = [q['sql'] for q in cq.captured_queries if re_from_ds.search(q['sql'])]
```

Comment:
NIT: This is one liner took me a bit of staring to grok, might be nice to break it up a bit for readability.

Reply:
Sorry, my Lisp is leaking. :-)

Reply:
I made it more explicit, although I'm not sure how much the readability is improved.